contracts/pool/src/contract.rs

```diff
@@ -415,12 +415,14 @@ impl LiquidityPoolTrait for LiquidityPool {
         let pool_balance_a = utils::get_pool_balance_a(&env);
         let pool_balance_b = utils::get_pool_balance_b(&env);
 
-        let mut share_ratio = Decimal::zero();
         let total_shares = utils::get_total_shares(&env);
-        if total_shares != 0i128 {
-            share_ratio = Decimal::from_ratio(share_amount, total_shares);
+
+        if total_shares == 0i128 {
+            panic_with_error!(env, ContractError::TotalSharesEqualZero);
```

Comment:
1) Stay consistent with the rest of the code base.
2) Notice, that the auditor reported a very important problem - if in this place total shares are equal to 0, it's a critical issue within the whole protocol. It should be exposed more.
```suggestion
        log!(
            &env,
            "Pool: WithdrawLiquidity: Critical error - Total shares are equal to zero before withdrawal!"
        );
        if total_shares == 0i128 {
            panic_with_error!(env, ContractError::TotalSharesEqualZero);
```